Added FILTER_VALIDATE_BOOL

sharkwouter opened the pull request:
This makes the filter names a little bit more consistent. Currently there are the following filters:

- FILTER_VALIDATE_BOOLEAN
- FILTER_VALIDATE_INT

This makes the user expect they can use either FILTER_VALIDATE_BOOL or FILTER_VALIDATE_INTEGER as well. This can be seen in this reddit thread: https://www.reddit.com/r/lolphp/comments/ebgewe/int_is_short_for_integer_bool_is_short_for/

This commit doesn't introduce new filters, just additional names for the mentioned two filters.

I've tested it with the following script:
```
<?php
$int = 1;
$string = "hallo";
$boolean = TRUE; 
print("output: " . filter_var($int, FILTER_VALIDATE_INT) . "\n");
print("output: " . filter_var($int, FILTER_VALIDATE_INTEGER) . "\n");
print("output: " . filter_var($string, FILTER_VALIDATE_INT) . "\n");
print("output: " . filter_var($string, FILTER_VALIDATE_INTEGER) . "\n");

print("output: " . filter_var($boolean, FILTER_VALIDATE_BOOL) . "\n");
print("output: " . filter_var($boolean, FILTER_VALIDATE_BOOLEAN) . "\n");
print("output: " . filter_var($string, FILTER_VALIDATE_BOOL) . "\n");
print("output: " . filter_var($string, FILTER_VALIDATE_BOOLEAN) . "\n");
```

Output:
```
output: 1
output: 1
output: 
output: 
output: 1
output: 1
output: 
output: 
```

nikic commented:
Long term, we should not have aliases for these filters. I would suggest to instead frame this as "Rename FILTER_VALIDATE_BOOLEAN to use the canonical type name FILTER_VALIDATE_BOOL" and slate FILTER_VALIDATE_BOOLEAN for eventual deprecation.

In particular I don't think we should be adding non-canonical FILTER_VALIDATE_INTEGER. We'd also have to add FILTER_VALIDATE_DOUBLE and possibly FILTER_VALIDATE_REAL if we go down that line.

KalleZ commented:
I agree with @nikic here, it should be either-or, similar to how we dealt with the recent MAGIC_QUOTES > ADD_SLASHES filter

sharkwouter commented:
That would be a good solution as well. I'll look into renaming the FILTER_VALIDATE_BOOLEAN function and I'll remove the FILTER_VALIDATE_INTEGER constant then. I'll be able to do so and test it later today,

sharkwouter commented:
I've renamed FILTER_VALIDATE_BOOLEAN to FILTER_VALIDATE_BOOL now and removed the FILTER_VALIDATE_INTEGER constant. FILTER_VALIDATE_BOOLEAN can still be used, but refers to FILTER_VALIDATE_BOOL.

I ran ``make test`` again and I'm not seeing any issues. Perhaps tests making use of FILTER_VALIDATE_BOOLEAN should be changed to use FILTER_VALIDATE_BOOL instead later.

I also tested with this script:
```
<?php
$string = "hallo";
$boolean = TRUE; 
print("output: " . filter_var($boolean, FILTER_VALIDATE_BOOL) . "\n");
print("output: " . filter_var($boolean, FILTER_VALIDATE_BOOLEAN) . "\n");
print("output: " . filter_var($string, FILTER_VALIDATE_BOOL) . "\n");
print("output: " . filter_var($string, FILTER_VALIDATE_BOOLEAN) . "\n");
```
This gave the expected output:
```
output: 1
output: 1
output: 
output: 
```

Renaming the ``php_filter_boolean`` function could be done as well, but I'm not sure if that would make sense.

nikic commented:
Is it possible to also reference filters by their name (`"boolean"`) rather than the constant? I'm wondering if we need to register both for that or not.

carusogabriel commented:
> I also tested with this script:

@sharkwouter Do you mind add a proper `.phpt` test for this change?

nikic commented:
It looks like the filter names are used by filter_id() and filter_list(). I'm not totally clear on whether or not it's fine to simply change a filter name, though I'd tend to say that it's okay.

sharkwouter commented:
> > I also tested with this script:
> 
> @sharkwouter Do you mind add a proper `.phpt` test for this change?

Will do!

> It looks like the filter names are used by filter_id() and filter_list(). I'm not totally clear on whether or not it's fine to simply change a filter name, though I'd tend to say that it's okay.

I'll look into it.

sharkwouter commented:
I've added a test. For this I've copied the test 014.phpt, which tests FILTER_VALIDATE_BOOL, and changed all instances of FILTER_VALIDATE_BOOLEAN. I wasn't sure what to call it, so I called it 061.phpt. Let me know if it needs a different name.

I've tested filter_id() and filter_list() as well, but they don't work in a different way from before. They use the string "boolean" to represent this filter. Using something like ``filter_var("no", filter_id("boolean")`` works like expected.

cmb69 commented:
Thanks for checking @sharkwouter! In my opinion, this test is not necessary (since it's just an alias); having it shouldn't hurt though (and I don't mind the name of the file).

nikic commented:
Squashed and merged as https://github.com/php/php-src/commit/f13d0a72d5cf92785c91ffc33c27df3df3f8e96e. Thanks!
